### DISCUSSION TITLE
Can we switch to Pan/Rotation using a key and keep them selected ?

### DISCUSSION THREAD

**Discussion by oussamabonnor1** [iTwin/itwinjs-core#3189] [Q&A] [answered]
I understand that clicking `Shift + Mouse Wheel` activates Rotation and using `Mouse Wheel` alone activates the Pan. But once released, the tool goes back to the Selection Tool by default.

Is there a way to select Rotation or Pan using a key to get a result similar to selecting them from the Navbar tools.

![image](https://user-images.githubusercontent.com/17766221/153185020-c6be103b-2377-4891-adc0-0c12dec23fec.png)

**Comment by bbastings** [comment]
You can define keyboard shortcuts to run commands using ConfigurableUiManager.loadKeyboardShortcuts: [shortcut](https://www.itwinjs.org/reference/appui-react/configurableui/configurableuimanager/loadkeyboardshortcutsstatic/)

**Reply by oussamabonnor1** [reply]
Looks great, how about actually selecting the tools, is there an event for that ?

**Reply by bbastings** [reply]
Selecting how, from the toolbar? There's an event for when a new tool is started: [tool change event](https://www.itwinjs.org/reference/core-frontend/tools/tooladmin/activetoolchanged/)

**Reply by bsteinbk** [reply]
If you are using a UiItemsProvider to provide these tools to the horizontal toolbar, you should be using an entry like [this](https://github.com/iTwin/itwinjs-core/blob/aecec84a91f331aa940700ee9db6a101cd7119e0/ui/appui-react/src/appui-react/ui-items-provider/StandardNavigationToolsProvider.ts#L61) to define the button. This will allow the button to be associated with its tool via its ToolId. If that is done then the button will show as selected when the tool is active.

**Reply by oussamabonnor1** [reply]
I think I was not clear, sorry. I want to click Shift on my keyboard & switch between Rotation/Pan using it. Without clicking on a UI button.

**Reply by bsteinbk** [reply]
@bbastings suggestion to use the keyboard shortcuts to activate the appropriate tool is what you should be looking at. I was merely stating that if you want the tool buttons to reflect what tool is active, they need to be properly defined.

**Comment by oussamabonnor1** [comment, marked as answer]
Had to get back here since we solved this, we used:

```typescript
//toolId can be RotateViewTool.toolId for Rotation or  PanViewTool.toolId for Panning.
IModelApp.tools.run(toolId, args);
```
